runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/GreedyStageFuser.java

```diff
@@ -141,6 +149,55 @@ public static ExecutableStage forGrpcPortRead(
         materializedPCollections);
   }
 
+  /**
+   * Patches all input coders of stateful ExecutableStage transforms. Stateful transforms aways have
+   * a KvCoder as input. Flink partitions the data based on the serialized version of the key. This
```

Comment:
This comment is too specific to Flink and could be generalized to:
```
Patches all input coders of stateful Executable transforms to be able to ensure that the byte representation for the key matches the keys byte representation received for state requests. This is necessary to allow runners to be able to match the input key to the state request key to be able to:
* fetch and store user state
* fetch and store timers
```

Reply:
Have adapted the comment to remove the Flink context.